packages/client/src/events.ts

```diff
@@ -0,0 +1,80 @@
+import EventEmitter3, { EventNames, EventListener } from 'eventemitter3'
+import { StorageNodeAssignmentEvent } from './StorageNodeRegistry'
+
+export interface StreamrClientEvents {
+    addToStorageNode: (payload: StorageNodeAssignmentEvent) => void,
+    removeFromStorageNode: (payload: StorageNodeAssignmentEvent) => void
+}
+
+interface ObservableEventEmitterEvents {
+    addEventListener: (eventName: keyof StreamrClientEvents) => void
+    removeEventListener: (eventName: keyof StreamrClientEvents) => void
+}
+
+/*
+ * An observable EventEmitter: emits an addEventListener/removeEventListener event when a listener
+ * is added or removed
+ */
+export class StreamrClientEventEmitter<C = any> extends EventEmitter3<StreamrClientEvents & ObservableEventEmitterEvents> {
+
+    on<T extends EventNames<StreamrClientEvents & ObservableEventEmitterEvents>>(
+        event: T, fn: EventListener<StreamrClientEvents & ObservableEventEmitterEvents, T>, context?: C
+    ) {
+        super.on(event, fn, context)
+        this.emitListenerEvent('addEventListener', event)
+        return this
+    }
+
+    once<T extends EventNames<StreamrClientEvents & ObservableEventEmitterEvents>>(
+        event: T, fn: EventListener<StreamrClientEvents & ObservableEventEmitterEvents, T>, context?: C
+    ) {
+        const wrappedFn: any = (...args: any[]) => {
+            (fn as any).apply(context, args)
+            this.emitListenerEvent('removeEventListener', event)
```

Comment:
I think it may be more clear if the "meta"-events (so to say) wouldn't be events at all but instead implemented as callback functions.

Reply:
Alternatively have "meta"-event be emitted and listened on a separate eventemitter instance.